### DISCUSSION TITLE
Question: Import location

### DISCUSSION THREAD

**Discussion by rastaman111** [TortugaPower/BookPlayer#632] [Q&A] [answered]
@GianniCarlo hi!

I have implemented your import in my project and faced this problem.  I created a folder, go to it and want to create a file in it, and it is created in the root daddy, where is the definition of where to create the file?

**Comment by GianniCarlo** [comment]
I'm moving this to Discussions

**Comment by GianniCarlo** [comment, marked as answer]
Not really sure what you mean, but to clarify, the `ImportOperation` only moves the folder from the source (usually Inbox or Documents folder, sometimes a tmp folder), to the destination `Processed` folder, which contains all the imported files. An imported file should always have a corresponding element in the database. After that, the interaction is shown to the user if they want to move the items elsewhere (if there's more than one), to either a new/existing playlist or keep in the library

**Reply by rastaman111** [reply]
yes, I understood this, when I transfer the file to the application, a window opens where this file is displayed and then it appears in the Processed folder (root).  but I created a folder in this directory, for example NF, go to it and transfer the file to the application, open Import and the new file is sent to Processed, and not to NF

**Reply by rastaman111** [reply]
https://vimeo.com/601173426

link to the video of my problem, let me know by video availability

**Reply by GianniCarlo** [reply]
Look into the function `handleOperationCompletion`, the `LibraryViewController` should be checking if it has a `PlaylistViewController` being shown, and if so, will pass the operation completion callback to that controller. Maybe that validation is failing 🤔

**Reply by rastaman111** [reply]
> Look into the function `handleOperationCompletion`, the `LibraryViewController` should be checking if it has a `PlaylistViewController` being shown, and if so, will pass the operation completion callback to that controller. Maybe that validation is failing 🤔 

I looked where you said and yes I forgot to indicate it at myself, now everything works
